lib/commands/misc/giverole.js

```diff
@@ -24,46 +24,80 @@ module.exports = class GiveRoleCommand extends Command {
 		this.queues = {};
 	};
 
+	grantRole(msg, role, channel) {
+		msg.member.addRole(role);
+		channel.send(`${msg.author} now has the ${role.name} role!`);
+	}
+
+	giveMemberRole(msg, role, channel, userRoles) {
+		// Only allows this to run if the user is not already a Member
+		if (!userRoles.has(role.id)) {
+			msg.member.addRole(role);
+
+			channel.send(`Welcome to the server, ${msg.author}! <:wutGiga:297897855727697921>`);
+		}
+	}
+
+	giveColorRole(msg, role, channel, userRoles, memberRole) {
+		// Confirms that the user doesn't already have the color role and is a Member
+		if (!userRoles.has(role.id) && userRoles.has(memberRole.id)) {
+
+			// Prevents the user from claiming more than one color role
+			if (!userRoles.some(r => COLOR_ROLES.includes(r.name))) {
+				this.grantRole(msg, role, channel);
+			} else {
+				msg.reply("You have already claimed another color role! Use r!rmrole [role name] and try again.");
+			}
+
+		} else {
+		  msg.reply("You have already claimed that role! Use r!rmrole [role name] and try again.");
```

Comment:
Your if condition checks that "the user doesn't already have a color role" and "is a Member", but the message only caters for the first case! Let's change the if statement. You can also look into checking for "is a Member" before calling this method.